### PR TITLE
Adding edge case handling for string inputs into norm

### PR DESCRIPTION
**Pull request by KerimovEmil**
Signed-off-by: KerimovEmil 

## Description
Raising an error for non-supported string arguments into norm atom, and handling the fro norm.
Addressing issue: https://github.com/cvxpy/cvxpy/issues/1865

## Type of change
- [ ] New feature (backwards compatible)
- [ ] New feature (breaking API changes)
- [x] Bug fix
- [ ] Other (Documentation, CI, ...)

## [Contribution checklist](https://www.cvxpy.org/contributing/index.html#contribution-checklist)
- [x] Add our license to new files.
- [x] Check that your code adheres to our coding style.
- [x] Write unittests.
- [x] Run the unittests and check that they’re passing.
- [x] Run the benchmarks to make sure your change doesn’t introduce a regression.

### CODE REVIEW

**Comment by phschiele** [comment]
@KerimovEmil thanks for taking the time to make this PR!
Generally, for a bug fix, it is nice to provide a test case that would fail before your PR, and pass after.
In your case, you can add a test `test_norm_exceptions()` to `test_atoms.py`.
You can check `test_list_input()` in the same file for an example of such an exception test.

**Comment by KerimovEmil** [comment]
> @KerimovEmil thanks for taking the time to make this PR!
> Generally, for a bug fix, it is nice to provide a test case that would fail before your PR, and pass after.
> In your case, you can add a test `test_norm_exceptions()` to `test_atoms.py`.
> You can check `test_list_input()` in the same file for an example of such an exception test.

Thanks! I'll try to add that later tonight.

File: cvxpy/atoms/norm.py [hunk 1]
```diff
@@ -69,6 +69,10 @@ def norm(x, p: Union[int, str] = 2, axis=None):
             return norm1(x, axis=axis)
         elif p in [np.inf, "inf", "Inf"]:
             return norm_inf(x, axis)
+        elif p == 'fro':
```
Comment (phschiele) [inline cvxpy/atoms/norm.py:72]:
```suggestion
        elif str(p).lower == "inf":
            return norm_inf(x, axis)
        elif str(p).lower == "fro":
```
`str(np.inf) == "inf"`

**Comment by SteveDiamond** [comment]
Looks good, thanks @KerimovEmil !

**Pull request merged**